equational_theories/Generated/All4x4Tables/src/generate_lean.py

```diff
@@ -146,22 +159,47 @@ def «{name}» : Magma (Fin {div}) where
 """
     return out
 
-
-with open(f"{dir}/data/refutations.txt") as f:
-    with open(f"{dir.parent}/All4x4Tables.lean", "w") as main:
-      lines = f.readlines()
-      # the format is groups-of-three-lines-based
-      lines = [lines[i:i + 3] for i in range(0, len(lines), 3)]
-      for i, line in enumerate(lines):
-          leanfile = f"{dir}/Refutation{i}.lean"
-          data = parse_row(line)
-          if data and data["div"] < 6:
-            data = prune_row(data)
-            print(f"Writing {leanfile}")
-            main.write(f"import equational_theories.Generated.All4x4Tables.Refutation{i}\n")
-            with open(leanfile, "w") as f:
-                  f.write(generate_lean(data))
-
+def create_rows(f):
```

Comment:
The logic is now spread (even more) in `create_row` and `parse_row`. Worth cleaning up, or are we going to make all tools use the same output format anyways?

Reply:
In the long run, the upcoming json format will replace all of the ad-hoc formats developed here.
But this is a good point, I'll have a quick think and see if I can push all the case analysis into create_row.

Reply:
Well, I tried, but I was not really satisfied with the outcome. The rewrite ended up being 6 lines longer, and imho harder to read.
I say let's kick this can down the road :)